Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM golang:1.17.3
+
+ADD . cloud-api-adaptor
+RUN git clone -b ccv0-peer-pods https://github.com/bpradipt/kata-containers
+RUN apt-get update && apt-get install -y libvirt-dev libvirt0 genisoimage # other requirements?
+WORKDIR cloud-api-adaptor
+# generic build
+ENV CLOUD_PROVIDER=aws
+RUN make
+RUN install cloud-api-adaptor /usr/local/bin/cloud-api-adaptor
+RUN make clean
+# libvirt build
+ENV CLOUD_PROVIDER=libvirt
```

Comment:
How about adding a Makefile target for cloud-api-adaptor `docker-build` and `docker-push` with `CLOUD_PROVIDER` as a build time ARG. ? This will be more flexible imho